File: app/src/main/java/igrek/forceawaken/domain/alarm/AlarmTrigger.java
package igrek.forceawaken.domain.alarm;

import android.app.PendingIntent;
import android.os.Parcel;
import android.os.Parcelable;

import org.joda.time.DateTime;

public class AlarmTrigger implements Parcelable {
	
	private DateTime triggerTime;
	private boolean active;
	private PendingIntent pendingIntent;
	
	public AlarmTrigger(DateTime triggerTime, boolean active, PendingIntent pendingIntent) {
		this.triggerTime = triggerTime;
		this.active = active;
		this.pendingIntent = pendingIntent;
	}
	
	public DateTime getTriggerTime() {
		return triggerTime;
	}
	
	public boolean isActive() {
		return active;
	}
	
	public void setActive(boolean active) {
		this.active = active;
	}
	
	public PendingIntent getPendingIntent() {
		return pendingIntent;
	}
	
	@Override
	public String toString() {
		return triggerTime.toString("HH:mm:ss, yyyy-MM-dd");
	}
	
	@Override
	public boolean equals(Object obj) {
		return obj instanceof AlarmTrigger && triggerTime.equals(((AlarmTrigger) obj).triggerTime);
	}
	
	@Override
	public int describeContents() {
		return 0;
	}
	
	@Override
	public void writeToParcel(Parcel dest, int flags) {
		dest.writeSerializable(triggerTime);
		dest.writeByte((byte) (active ? 1 : 0));
		dest.writeParcelable(pendingIntent, flags);
	}
	
	private AlarmTrigger(Parcel in) {
		this.triggerTime = (DateTime) in.readSerializable();
		this.active = in.readByte() != 0;
		this.pendingIntent = in.readParcelable(PendingIntent.class.getClassLoader());
	}
	
	public static final Parcelable.Creator<AlarmTrigger> CREATOR = new Parcelable.Creator<AlarmTrigger>() {
		public AlarmTrigger createFromParcel(Parcel in) {
			return new AlarmTrigger(in);
		}
		
		public AlarmTrigger[] newArray(int size) {
			return new AlarmTrigger[size];
		}
	};
}

File: app/src/main/java/igrek/forceawaken/alarm/AlarmTrigger.kt
package igrek.forceawaken.alarm

import android.app.PendingIntent
import android.os.Parcel
import android.os.Parcelable
import org.joda.time.DateTime


data class AlarmTrigger(
        var triggerTime: DateTime,
        var isActive: Boolean,
        var pendingIntent: PendingIntent?,
) : Parcelable {

    override fun toString(): String {
        return triggerTime.toString("HH:mm:ss, yyyy-MM-dd")
    }

    override fun equals(obj: Any?): Boolean {
        return obj is AlarmTrigger && triggerTime == obj.triggerTime
    }

    companion object {
        @JvmField
        val CREATOR = object : Parcelable.Creator<AlarmTrigger> {
            override fun createFromParcel(parcel: Parcel): AlarmTrigger {
                val triggerTime = parcel.readSerializable() as DateTime
                val isActive = parcel.readByte().toInt() != 0
                val pendingIntent = parcel.readParcelable(PendingIntent::class.java.classLoader) as PendingIntent?
                return AlarmTrigger(triggerTime, isActive, pendingIntent)
            }

            override fun newArray(size: Int) = arrayOfNulls<AlarmTrigger>(size)
        }
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeSerializable(triggerTime)
        dest.writeByte((if (isActive) 1 else 0).toByte())
        dest.writeParcelable(pendingIntent, flags)
    }

    override fun describeContents() = 0
}